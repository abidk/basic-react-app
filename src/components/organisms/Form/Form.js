import React, { Component } from "react";
import Button from '../../atoms/Button';

class Form extends Component {

  constructor() {
    super();
    this.state = {
      name: ""
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ name: event.target.value });
  }

  handleSubmit(event) {
    const { name } = this.state;

    this.props.addName(name);

    this.setState({ name: "" });
  }

  render() {
    const { name } = this.state;
    return (
      <div className="form-group">
        {/* TODO atoms */}
        <label htmlFor="name">Name</label>
        <input type="text" id="name" value={name} onChange={this.handleChange} />

        <Button action={this.handleSubmit} label="Save" />
      </div>
    );
  }
}

export default Form;