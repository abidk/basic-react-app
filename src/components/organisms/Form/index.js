import { connect } from "react-redux";
import { addName } from "../../../state/actions";
import Form from './Form'

function mapDispatchToProps(dispatch) {
    return {
        addName: payload => dispatch(addName(payload))
    };
  }

export default connect(null, mapDispatchToProps)(Form);