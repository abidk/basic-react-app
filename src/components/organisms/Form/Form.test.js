import React from 'react'
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Form from './Form';

describe(__filename, () => {

    it('Should render', () => {    
        const output = mount(<Form />);

        expect(shallowToJson(output)).toMatchSnapshot();
    });
});