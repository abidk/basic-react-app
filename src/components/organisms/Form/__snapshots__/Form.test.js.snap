// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/abidkhalil/Documents/basic-react-app/src/components/organisms/Form/Form.test.js Should render 1`] = `
<Form>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <input
      id="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <Button
      action={[Function]}
      label="Save"
    >
      <button
        className="Button"
        onClick={[Function]}
      >
        Save
      </button>
    </Button>
  </div>
</Form>
`;
